torch/csrc/jit/codegen/cuda/transform_view.cpp

```diff
@@ -0,0 +1,678 @@
+  TrivialReductionTransform(const ViewIndexState& state)
+      : Transform(computeIndex(state)) {}
+
+  void toString(std::stringstream& output) const override {
+    output << "1-Red Index: " << index_ << std::endl;
+  }
+
+ private:
+  size_t computeIndex(const ViewIndexState& state) override {
+    return state.original_view_index;
+  }
+};
+
+//! The primary class that generates the transformations to go from
+//! the original view to the new view.
+class AnalyzeViewTransformation {
+ public:
+  AnalyzeViewTransformation(
+      const std::vector<IterDomain*> root_domain,
+      const std::vector<size_t>& original_view,
+      const std::vector<size_t>& new_view)
+      : root_domain_(root_domain),
+        original_view_(original_view),
+        new_view_(new_view),
+        transform_view_(original_view) {
+    // Check that the product of original and new view sizes are equal.
+    const size_t kOriginalNumElements = std::accumulate(
+        original_view_.begin(), original_view_.end(), 1, std::multiplies<>());
+    const size_t kNewNumElements = std::accumulate(
+        new_view_.begin(), new_view.end(), 1, std::multiplies<>());
+    TORCH_INTERNAL_ASSERT(kOriginalNumElements == kNewNumElements);
+  }
+
+  AnalyzeViewResult run() {
+    find_transformation();
+    TORCH_INTERNAL_ASSERT(
+        validate(),
+        "Analyze View Transformation failed to find valid transformation.\n",
+        toString());
+    return {
+        !broadcast_transforms_.empty(),
+        generateBroadcastAxes(),
+        generateTrivialReductionAxes(),
+        view_transforms_};
+  }
+
+ private:
+  std::vector<bool> generateBroadcastAxes() {
+    std::vector<bool> broadcast_axes(new_view_.size(), false);
+    for (auto bcast : broadcast_transforms_) {
+      broadcast_axes.at(bcast->index()) = true;
+    }
+    return broadcast_axes;
+  }
+
+  std::vector<int> generateTrivialReductionAxes() {
+    std::vector<int> reduction_axes;
+    for (auto tred : trivial_reduction_transforms_) {
+      reduction_axes.push_back(tred->index());
+    }
+    return reduction_axes;
+  }
+
+  std::string toString() {
+    std::stringstream output;
+    output << "===============================" << std::endl;
+    output << "old:";
+    for (auto s : original_view_) {
+      output << " " << s;
+    }
+    output << std::endl;
+
+    output << "===============================" << std::endl;
+    output << "new:";
+    for (auto s : new_view_) {
+      output << " " << s;
+    }
+    output << std::endl;
+
+    output << "===============================" << std::endl;
+    for (auto move : trivial_reduction_transforms_) {
+      move->toString(output);
+    }
+    for (auto move : view_transforms_) {
+      move->toString(output);
+    }
+    for (auto move : broadcast_transforms_) {
+      move->toString(output);
+    }
+    output << "===============================" << std::endl;
+    return output.str();
+  }
+
+  void addMergeTransform(bool lhs, bool rfactor) {
```

Comment:
The order of the two parameters is opposite of `AnalyzeMergeInterface::handle`, which would make the code prone to errors